pkg/sql/logictest/testdata/logic_test/alter_primary_key

```diff
@@ -1397,23 +1397,6 @@ a  b  k
 statement ok
 ALTER TABLE t ALTER PRIMARY KEY USING COLUMNS (k);
 
-# Legacy schema changer and declarative schema changer will produce slightly
-# different secondary index ordering for alter primary key, hence we assert
-# separately with slightly different expected output.
-onlyif config local-legacy-schema-changer
-query T
-SELECT create_statement FROM [SHOW CREATE TABLE t];
-----
-CREATE TABLE public.t (
-  a INT8 NOT NULL,
-  b INT8 NOT NULL,
-  k INT8 NOT NULL AS (a + b) VIRTUAL,
-  CONSTRAINT t_pkey PRIMARY KEY (k ASC),
-  UNIQUE INDEX t_a_key (a ASC),
-  INDEX t_idx_b_k (b ASC, k ASC)
-)
-
-skipif config local-legacy-schema-changer
```

Comment:
This is an example where previously this slight difference required us to write two expected output for the same thing (one for legacy schema changer and another for declarative schema changer). This is very ugly and I'm glad we can get rid of them now.